pkg/controlplane/control/manager.go

```diff
@@ -81,7 +87,37 @@ type Manager struct {
 	lock            sync.Mutex
 	serviceToImport map[string]types.NamespacedName
 
-	logger *logrus.Entry
+	podLock sync.RWMutex
+	// endpoints stores the IPv4/v6 address of cl-dataplane endpoints per namespace
+	endpoints map[string][]discv1.Endpoint
+	logger    *logrus.Entry
+}
+
+func (m *Manager) addImportEndpointSlice(ctx context.Context, imp *v1alpha1.Import) error {
+	m.logger.Infof("Adding import endpointslice '%s/%s'.", imp.Namespace, imp.Name)
+
+	protocol := v1.ProtocolTCP
+	port32 := int32(imp.Spec.TargetPort)
+
+	es := discv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imp.Name + "-" + uuid.NewString()[:5],
+			Namespace: imp.Namespace,
+			Labels: map[string]string{
+				labelServiceName: imp.Name,
+				labelESManagedBy: appName,
+			},
+		},
+		AddressType: discv1.AddressTypeIPv4,
+		Endpoints:   m.endpoints[imp.Namespace],
+	}
+	endpointPort := discv1.EndpointPort{
+		Port:     &port32,
+		Protocol: &protocol,
+	}
+	es.Ports = make([]discv1.EndpointPort, 1)
+	es.Ports[0] = endpointPort
```

Comment:
Can you move this to be part of the `es := ` assignment above?

Reply:
Sure.